Source/Scene/MetadataType.js

```diff
@@ -0,0 +1,285 @@
+import Check from "../Core/Check.js";
+import DeveloperError from "../Core/DeveloperError.js";
+import FeatureDetection from "../Core/FeatureDetection.js";
+
+/**
+ * An enum of metadata types.
+ *
+ * @exports MetadataType
+ *
+ * @private
+ */
+var MetadataType = {
+  INT8: "INT8",
+  UINT8: "UINT8",
+  INT16: "INT16",
+  UINT16: "UINT16",
+  INT32: "INT32",
+  UINT32: "UINT32",
+  INT64: "INT64",
+  UINT64: "UINT64",
+  FLOAT32: "FLOAT32",
+  FLOAT64: "FLOAT64",
+  BOOLEAN: "BOOLEAN",
+  STRING: "STRING",
+  ENUM: "ENUM",
+  ARRAY: "ARRAY",
+};
+
+/**
+ * Gets the minimum value for the numeric type.
+ * <p>
+ * Returns a BigInt for the INT64 and UINT64 types if BigInt is supported on this platform.
+ * Otherwise an approximate number is returned.
+ * </p>
+ *
+ * @param {MetadataType} type The type.
+ * @returns {Number|BigInt} The minimum value.
+ *
+ * @exception {DeveloperError} type must be a numeric type
+ */
+MetadataType.getMinimum = function (type) {
+  //>>includeStart('debug', pragmas.debug);
+  if (!MetadataType.isNumericType(type)) {
+    throw new DeveloperError("type must be a numeric type");
+  }
+  //>>includeEnd('debug');
+
+  switch (type) {
+    case MetadataType.INT8:
+      return -128;
+    case MetadataType.UINT8:
+      return 0;
+    case MetadataType.INT16:
+      return -32768;
+    case MetadataType.UINT16:
+      return 0;
+    case MetadataType.INT32:
+      return -2147483648;
+    case MetadataType.UINT32:
+      return 0;
+    case MetadataType.INT64:
+      if (FeatureDetection.supportsBigInt()) {
+        return BigInt("-9223372036854775808"); // eslint-disable-line
+      }
+      return -Math.pow(2, 63);
+    case MetadataType.UINT64:
+      if (FeatureDetection.supportsBigInt()) {
+        return BigInt(0); // eslint-disable-line
+      }
+      return 0;
+    case MetadataType.FLOAT32:
+      return -340282346638528859811704183484516925440.0;
```

Comment:
Add a comment saying that this can't be expressed in a simpler way in JS

Can you get the minimum value from a Float32Array?

Reply:
I couldn't find a way to get this in another way. The alternative is using the formula (2 − 2²³) × 2¹²⁷.